torchgeo/trainers/regression.py

```diff
@@ -26,36 +26,70 @@
 
 
 class RegressionTask(pl.LightningModule):
-    """LightningModule for training models on regression datasets."""
+    """LightningModule for training models on regression datasets.
+
+    Supports any available `Timm model
+    <https://rwightman.github.io/pytorch-image-models/>`_
+    as an architecture choice.
+    """
 
     def config_task(self) -> None:
         """Configures the task based on kwargs parameters."""
-        model = self.hyperparams["model"]
-        pretrained = self.hyperparams["pretrained"]
-
-        if parse(torchvision.__version__) >= parse("0.13"):
-            if pretrained:
-                kwargs = {
-                    "weights": getattr(
-                        torchvision.models, f"ResNet{model[6:]}_Weights"
-                    ).DEFAULT
-                }
+        in_channels = self.hyperparams["in_channels"]
+        regression_model = self.hyperparams["regression_model"]
```

Comment:
Currently there's a mix of `<task>_model` and `model` used throughout TorchGeo. I think it would be better to use `model` and `backbone`. We may also want to divide up the docs so that datamodules are listed by task and not by geo/nongeo. Maybe same for datasets too, haven't decided. But the docs can wait. Just wanted to mention the kwarg thing now since you're changing it.

Reply:
do you mean the argument `<task_model>` should be named just `model` for classification and regression and then instead of `self.model` and everywhere else we mention `model` like in the documentation, it should actually say `backbone`?

Reply:
What I'm suggesting is that `model` should refer to the name of the overall model, while `backbone` refers to the model backbone (if one exists). So for classification and regression, there would only be a `model` parameter (e.g., ResNet50), while for segmentation there would be a `model` (e.g., UNet) and a `backbone` (e.g., ResNet50).

Although SMP seems to call them encoder/decoder, not backbone. So maybe it would depend on the model?